pkg/client/clients.go

```diff
@@ -67,7 +71,8 @@ func (c *Clients) WaitForSync(interval time.Duration, timeout time.Duration) err
 			c.PVInformer.Informer().HasSynced() &&
 			c.StorageInformer.Informer().HasSynced() &&
 			c.ConfigMapInformer.Informer().HasSynced() &&
-			c.NamespaceInformer.Informer().HasSynced()
+			c.NamespaceInformer.Informer().HasSynced() &&
+			c.ApplicationInformer.Informer().HasSynced()
```

Comment:
Seems this could be causing a NPE when app-CRD is not enabled.
Could you please add a nil check here?

Reply:
Good catch. This is valid. I will fix it